cleanslate/doma/views.py

```diff
@@ -204,3 +200,57 @@ def delete_chore(request, pk):
 
         return HttpResponseRedirect(reverse(reminders))
     return render(request, 'chore_delete_form.html', {})
+
+def create_user(request):
+    if request.method == 'POST':
+        form = CreateUserForm(request.POST)
+        if form.is_valid():
+            try:
+                new_user = User.objects.create_user(username = form.cleaned_data['username'], email = form.cleaned_data['email'], password = form.cleaned_data['password'])
+                messages.success(request, 'You successfully created a new user. Sign in now.')
+                return HttpResponseRedirect(reverse(profile))
+            except IntegrityError as e:
+                messages.error(request, "You have not met Django's built in attribute requirements. Try using a stronger password and a longer username.")
+                return render(request, 'form.html', {'form': form})
+        else:
+            messages.error(request, 'Please correct the errors in the form.')
+            return render(request, 'form.html', {'form': form})
+    else:
+        form = CreateUserForm()
+    return render(request, 'form.html', {'form': form})
+
+def edit_user(request, pk):
+    updated_user = User.objects.filter(pk = pk)[0]
+    if request.method == 'POST':
+        form = EditUserForm(request.POST)
+        if form.is_valid():
+            updated_user.username = form.cleaned_data['username']
+            updated_user.email = form.cleaned_data['email']
+            updated_user.set_password(form.cleaned_data['password'])
+            updated_user.save()
+            update_session_auth_hash(request, updated_user)
+            messages.success(request, 'You successfully updated your account settings.')
+            return HttpResponseRedirect(reverse(profile))
+        else:
+            messages.error(request, 'Please correct the errors in the form.')
+    else:
+        form = EditUserForm(initial = model_to_dict(updated_user))
+    return render(request, 'form.html', {'form': form})
+
+@login_required
+def create_home(request):
+    if request.method == 'POST':
+        form = CreateHomeForm(request.POST)
+        if form.is_valid():
+            new_home = Home.objects.create(name = "", address = "", created_by = request.user.profile)
```

Comment:
Okay, cool. You did a similar thing to what I did before. Wasn't sure if this was good to do.

Reply:
With a different form

Reply:
It's probably a little redundant but it works. It did cause me some problems with my @receivers for the forum later, but I hacked my way around that by saving the forum name a little later than I had initially. Your form code here was a life saver though, it would've taken me so much longer to learn forms had your code not been there, good job and thanks :+1: